guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,954 @@
+    pub last_name: String,
+    pub email: String,
+}
+
+#[derive(Deserialize, Insertable)]
+#[table_name="users"]
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    #[column_name(email)]
+    pub electronic_mail: &'a str,
+ }
+```
+
+```rust
+// File: src/main.rs
+#[macro_use] extern crate diesel;
+
+use diesel::prelude::*;
+
+fn main() {
+    // Here we simulate a webform by deserializing a json string
+    // into a NewUser struct.
+    let new_user: NewUser = serde_json::from_str(
+        r#"{
+            "first_name": "Gordon",
+            "last_name": "Freeman",
+            "electronic_mail": "gordon.freeman@blackmesa.co"
+        }"#).unwrap();
+
+    diesel::insert_into(users::table)
+        .values(&new_user)
+        .execute(&db_connection)
+        .expect("Error inserting row into database");
+
+    let all_users = users.load::<User>(&db_connection)
+        .expect("Error loading users");
+
+    println!("User count: {}", all_users.len());
+    //=> User count: 1
+}
+```
+
+If we try to insert records without deriving `Insertable`, we would get the following error.
+
+```rust
+error[E0277]: the trait bound `&diesel_demo::models::NewUser<'_>: diesel::Insertable<diesel_demo::schema::users::table>` is not satisfied
+  --> src/bin/main.rs:29:10
+   |
+29 |         .values(&new_user)
+   |          ^^^^^^ the trait `diesel::Insertable<diesel_demo::schema::users::table>` is not implemented for `&diesel_demo::models::NewUser<'_>`
+```
+
+The compiler reports that `Insertable` isn't implemented for our `NewUser` struct.
+
+## Identifiable
+
+Certain database operations,
+such as table [associations](#associations) and updates, 
+require that rows be uniquely identifiable.
+Implementing the `Identifiable` trait on your struct will define which columns (primary keys)
+make your struct uniquely identifiable.
+To implement `Identifiable`, 
+annotate your struct with [`#[derive(Identifiable)]`][identifiable_doc].
+
+[identifiable_doc]: https://docs.diesel.rs/diesel/associations/trait.Identifiable.html
+
+By default, `Identifiable` will assume the primary key is a column named `id`.
+If your table's primary key is named differently,
+you can annotate the table with the attribute `#[primary_key(some_field_name)` or `#[primary_key(some_field_name, another_field_name)`.
+The `Identifiable` trait will assume that the annotated struct will be named in the singular form of the table it corresponds to.
+If the table name differs you may use the `#[table_name="some_table_name"]` attribute annotation.
+The `Identifiable` trait gives us the `id()` method on our models,
+which returns the value of our record's primary key.
+
+In the following example,
+we will look at some of the behavior `Identifiable` gives us.
+We will add the annotation to our `User` struct.
+We will then attempt to get the value of the first record's primary key by calling `id()` and also update the first and last name of our user.
```

Comment:
These three lines begin with 'we will'… maybe use "First, we'll look", "After that, let's add", and "Finally, we will attempt"?